test/plural_resolver_test.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:i18next/i18next.dart';
+import 'package:i18next/src/plural_resolver.dart';
+import 'package:mockito/mockito.dart';
+
+abstract class FakeCheckerKeyFunction {
+  bool call(String pluralModifier);
+}
+
+class MockCheckerKeyFunction extends Mock implements FakeCheckerKeyFunction {}
+
+void main() {
+  group('PluralResolver', () {
+    MockCheckerKeyFunction checkerKeyFnc;
+    PluralResolver subject;
+
+    setUpAll(() {
+      checkerKeyFnc = MockCheckerKeyFunction();
+      subject = PluralResolver();
+    });
+
```

Comment:
Missing test when count is null

Reply:
IMHO I don't think we need to test (specifically for PluralResolver) when the count is null since we have this validation on the method `translateKey`

```dart
    final needsPlural = count != null;

    String pluralSuffix;
    if (needsPlural) {
      pluralSuffix = pluralResolver.pluralize(
        count,
        options,
        (pluralModifier) => _hasPluralKey(
          pluralModifier,
          baseKey: key,
          locale: locale,
          namespace: namespace,
          options: options,
        ),
      );
    }
``` 

Maybe it should be good to add this requested test to `Translator` class file tests, sound good?

Reply:
Yes, you're right.

Since the plural resolver is not public, then the only caller (translator) already does the initial checks/assertions early on.